File: 884-uncommon-words-from-two-sentences/884.py
class Solution:
    def uncommonFromSentences(self, s1: str, s2: str) -> List[str]:
        ans = []
        table = {}

        for s in (s1 + " " + s2).split():
            if s in table:
                table[s] += 1
            else:
                table[s] = 1

        for t in table:
            if table[t] == 1:
                ans.append(t)

        return ans

File: 884-uncommon-words-from-two-sentences/884.cpp
class Solution {
private:
    void split(const string& str, map<string, int> &m) {
        istringstream iss(str);
        string token;
        set<string> s;
        while (iss >> token) {
            if (!token.empty()) {
                ++m[token];
            }
        }
    }
public:
    vector<string> uncommonFromSentences(string s1, string s2) {
        map<string, int> m;
        vector<string> ans;
        split(s1, m);
        split(s2, m);
        for (const pair<string, int>& p : m) {
            if (p.second == 1)
                ans.push_back(p.first);
        }
        return ans;
    }
};
